src/network/receive.ts

```diff
@@ -496,7 +497,8 @@ export async function parseKeysendInvoice(
   i: interfaces.Invoice
 ): Promise<void> {
   try {
-    const hash = i.r_hash.toString('base64')
+    // const hash = i.r_hash.toString('base64')
+    const hash = sha.sha256.hex(JSON.stringify(i))
```

Comment:
@antonilol so this caching system was added for greenlight because for some reason it was showing duplicate messages, the PR and the issue is below

https://github.com/stakwork/sphinx-relay/issues/274
https://github.com/stakwork/sphinx-relay/pull/277

Reply:
Will this still catch the duplicates?

or does some id need to be compared

Reply:
Maybe a `if (IS_GREENLIGHT)` can be added to sort of 'say' in the code this is only needed for greenlight

Reply:
Yea we might just need to use `if (IS_GREENLIGHT)` since I don't think the amp changes will change greenlight as of now, but also if this works aswell we can just use this instead of having custom greenlight logic

Reply:
just need to test with greenlight aswell

Reply:
You guys are correct, that was only in there for Greenlight. I think adding an if(IS_GREENLIGHT) check is fine